pkg/k8sapi/k8sutils.go

```diff
@@ -26,25 +26,34 @@ import (
 )
 
 const (
-	awsNode = "aws-node"
+	awsNode             = "aws-node"
+	envEnablePodENI     = "ENABLE_POD_ENI"
+	envCustomNetworkCfg = "AWS_VPC_K8S_CNI_CUSTOM_NETWORK_CFG"
 )
 
 var log = logger.Get()
 
 // Get cache filters for IPAMD
 func getIPAMDCacheFilters() map[client.Object]cache.ByObject {
 	if nodeName := os.Getenv("MY_NODE_NAME"); nodeName != "" {
-		return map[client.Object]cache.ByObject{
+		filter := map[client.Object]cache.ByObject{
 			&corev1.Pod{}: {
 				Field: fields.Set{"spec.nodeName": nodeName}.AsSelector(),
 			},
 			&corev1.Node{}: {
 				Field: fields.Set{"metadata.name": nodeName}.AsSelector(),
 			},
-			&rcscheme.CNINode{}: {
+		}
+		// only cache CNINode when SGP or custom network is in use
+		enabledPodENI := utils.GetBoolAsStringEnvVar(envEnablePodENI, false)
+		enabledCustomNetworkCFG := utils.GetBoolAsStringEnvVar(envCustomNetworkCfg, false)
+		if enabledPodENI || enabledCustomNetworkCFG {
```

Comment:
Should this be `AND` since CNINode is only used when SGP is enabled at this moment.

Reply:
@haouc, looks like currently it's checking
1. if custom networking is enabled -> if SGP is enabled: add feature to CNINode
2. if SGP is enabled: add feature to CNINode
https://github.com/aws/amazon-vpc-cni-k8s/blob/master/pkg/ipamd/ipamd.go#L564
So essentially we can just check if SGP is enabled or not.